devito/passes/clusters/misc.py

```diff
@@ -53,17 +53,29 @@ def callback(self, clusters, prefix):
                 processed.append(c)
                 continue
 
-            # Scalars prevent lifting if they are read by another Cluster
-            swrites = {f for f in c.scope.writes if f.is_Symbol}
-            if any(swrites & set(i.scope.reads) for i in impacted):
-                processed.append(c)
-                continue
-
-            # Contract iteration and data spaces for the lifted Cluster
+            # The contracted iteration and data spaces
             key = lambda d: d not in hope_invariant
             ispace = c.ispace.project(key).reset()
             dspace = c.dspace.project(key).reset()
 
+            # All of the inner Dimensions must appear in the write-to region
```

Comment:
ok, this is helpful

Reply:
yeah, this is fixing a bug with lifting!